### ISSUE TITLE
Valgrind uninitialized error in test_math_utils 

### ISSUE THREAD

**Issue by pooyan-dadvand** [KratosMultiphysics/Kratos#1333]
Valgrind:

```Valgrind
==27932== Conditional jump or move depends on uninitialised value(s)
==27932==    at 0x8495BCD: Kratos::Testing::TestMathUtilsInvertMatrixTest::TestFunction() (test_math_utils.cpp:433)
==27932==    by 0x92585A7: Kratos::Testing::TestCase::Run() (test_case.cpp:54)
==27932==    by 0x925A8E1: Kratos::Testing::Tester::RunSelectedTestCases() (tester.cpp:266)
==27932==    by 0x7E1BFB6: invoke<int, void (*)()> (invoke.hpp:81)
==27932==    by 0x7E1BFB6: operator() (caller.hpp:223)
==27932==    by 0x7E1BFB6: boost::python::objects::caller_py_function_impl<boost::python::detail::caller<void (*)(), boost::python::default_call_policies, boost::mpl::vector1<void> > >::operator()(_object*, _object*) (py_function.hpp:38)
==27932==    by 0x9749F5C: boost::python::objects::function::call(_object*, _object*) const (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x974A157: boost::detail::function::void_function_ref_invoker0<boost::python::objects::(anonymous namespace)::bind_return, void>::invoke(boost::detail::function::function_buffer&) (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x9750902: boost::python::handle_exception_impl(boost::function0<void>) (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x97476F8: function_call (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x311646: PyObject_Call (in /usr/bin/python3.5)
==27932==    by 0x288EE0: PyEval_EvalFrameEx (in /usr/bin/python3.5)
==27932==    by 0x28D285: ??? (in /usr/bin/python3.5)
==27932==    by 0x28DF9E: PyEval_EvalCode (in /usr/bin/python3.5)
==27932==
```

and 

```Valgrind
==27932== Conditional jump or move depends on uninitialised value(s)
==27932==    at 0x8495BEF: Kratos::Testing::TestMathUtilsInvertMatrixTest::TestFunction() (test_math_utils.cpp:437)
==27932==    by 0x92585A7: Kratos::Testing::TestCase::Run() (test_case.cpp:54)
==27932==    by 0x925A8E1: Kratos::Testing::Tester::RunSelectedTestCases() (tester.cpp:266)
==27932==    by 0x7E1BFB6: invoke<int, void (*)()> (invoke.hpp:81)
==27932==    by 0x7E1BFB6: operator() (caller.hpp:223)
==27932==    by 0x7E1BFB6: boost::python::objects::caller_py_function_impl<boost::python::detail::caller<void (*)(), boost::python::default_call_policies, boost::mpl::vector1<void> > >::operator()(_object*, _object*) (py_function.hpp:38)
==27932==    by 0x9749F5C: boost::python::objects::function::call(_object*, _object*) const (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x974A157: boost::detail::function::void_function_ref_invoker0<boost::python::objects::(anonymous namespace)::bind_return, void>::invoke(boost::detail::function::function_buffer&) (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x9750902: boost::python::handle_exception_impl(boost::function0<void>) (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x97476F8: function_call (in /home/pooyan/Kratos/libs/libboost_python.so.1.59.0)
==27932==    by 0x311646: PyObject_Call (in /usr/bin/python3.5)
==27932==    by 0x288EE0: PyEval_EvalFrameEx (in /usr/bin/python3.5)
==27932==    by 0x28D285: ??? (in /usr/bin/python3.5)
==27932==    by 0x28DF9E: PyEval_EvalCode (in /usr/bin/python3.5)
==27932==
```

**Comment by loumalouomega** [comment]
Strange the conditional jump is just a conditional 👍 

~~~cpp
                    if (i == j) 
                    {
                        KRATOS_CHECK_NEAR(I(i,j), 1.0, tolerance);
                    }
                    else 
                    {
                        KRATOS_CHECK_NEAR(I(i,j), 0.0, tolerance);
                    }
~~~

**Comment by pooyan-dadvand** [comment]
It is in both of them...

**Comment by loumalouomega** [comment]
Solved in #1336

**Comment by loumalouomega** [comment]
I will close this
